benchmarks/gravity_prem/prem.cc

```diff
@@ -0,0 +1,176 @@
+/*
+  Copyright (C) 2011 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+#include <aspect/simulator.h>
+#include <aspect/material_model/simple.h>
+#include <aspect/boundary_velocity/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/vector_tools.h>
+
+namespace aspect
+{
+  namespace GravityPremBenchmark
+  {
+    using namespace dealii;
+
+    /**
+     * @note This benchmark only talks about the flow field, not about a
+     * temperature field. All quantities related to the temperature are
+     * therefore set to zero in the implementation of this class.
+     *
+     * @ingroup MaterialModels
+     */
+    template <int dim>
+    class PremDensity : public MaterialModel::Interface<dim>
+    {
+      public:
+        /**
+         * @name Physical parameters used in the basic equations
+         * @{
+         */
+        virtual void evaluate(const MaterialModel::MaterialModelInputs<dim> &in,
+                              MaterialModel::MaterialModelOutputs<dim> &out) const
+        {
+          for (unsigned int i=0; i < in.n_evaluation_points(); ++i)
+            {
+              const std::array<double,dim> position_point_sph = Utilities::Coordinates::cartesian_to_spherical_coordinates<dim>(in.position[i]);
+              const double x = position_point_sph[0] / 6371e3;
```

Comment:
easier to understand:
```suggestion
              const double normalized_r = position_point_sph[0] / 6371e3;
```